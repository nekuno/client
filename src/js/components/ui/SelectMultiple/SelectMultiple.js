import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Chip from '../Chip/Chip.js';
import styles from './SelectMultiple.scss';

export default class SelectMultiple extends Component {

    static propTypes = {
        title          : PropTypes.string,
        labels         : PropTypes.array.isRequired,
        values         : PropTypes.array.isRequired,
        onClickHandler : PropTypes.func.isRequired,
        color          : PropTypes.string,

    };

    handleClick(id) {
        this.props.onClickHandler(id);
    }


    onClickOptionHandler(event) {
        const {labels} = this.props;
        const value = event.target.value;

        if (labels.some(label => value == label.id)) {
            this.props.onClickHandler(value);
        }
    }

    render() {
        const {title, labels, values, color} = this.props;
        const showSelect = labels.length > 40;
        return (
            <div className={styles.selectMultiple}>
                {showSelect ?
                    <div className={styles.selectTag}>
                        <div className={styles.title + ' small'}>{title}</div>
                        <select onChange={this.onClickOptionHandler.bind(this)}>
                            <option key={'none'} value={''}>{}</option>
                            {labels.map(label => !values.some(value => value == label.id) ?
                                <option key={label.id} value={label.id}>{label.text}</option> : null
                            )}
                        </select>
                        {labels.map(label => values.some(value => value == label.id) ?
                            <Chip key={label.id} text={label.text} selected={true} color={color} onClickHandler={this.handleClick.bind(this, label.id)}/> : null
                        )}
                    </div>
                    :
                    <div className={styles.chips}>
                        {title ? <div className={styles.title + ' small'}>{title}</div> : null}
                        {labels.map(label => <Chip key={label.id} text={label.text} onClickHandler={this.handleClick.bind(this, label.id)} selected={values.some(value => value == label.id)}/>)}
                    </div>
                }
            </div>
        );
    }
}