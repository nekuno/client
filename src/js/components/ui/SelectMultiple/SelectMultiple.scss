@import '../../../../scss/_partials/colors';

.select-multiple {
  .select-tag {
    .title {
      font-weight: 600;
      margin-bottom: 12px;
      margin-left: 6px;
    }
    select {
      border-radius: 16px;
      -moz-border-radius: 16px;
      -webkit-border-radius: 16px;
      cursor: pointer;
      height: 35px;
      width: 100%;
      -webkit-appearance: none;
      -moz-appearance: none;
      -ms-appearance: none;
      appearance: none;
      option {
      }
    }
  }
  .chips {
    .title {
      font-weight: 600;
      margin-bottom: 12px;
      margin-left: 6px;
    }
  }
}