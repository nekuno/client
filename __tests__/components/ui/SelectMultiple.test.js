import React from 'react';
import SelectMultiple from '../../../src/js/components/ui/SelectMultiple/SelectMultiple.js';
import styles from '../../../src/js/components/ui/SelectMultiple/SelectMultiple.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const labels = [
    {
        id: "lorem",
        text: "Lorem"
    },
    {
        id: "ipsum",
        text: "Ipsum"
    },
    {
        id: "lorem-ipsum",
        text: "Lorem ipsum"
    },
    {
        id: "sed-ut-perspiciatis",
        text: "Sed ut perspiciatis"
    }
];

describe('Test SelectMultiple component', () => {
    const mockFn = jest.fn();
    const select = shallow(<SelectMultiple values={['lorem', 'lorem-ipsum']} labels={labels} title={'Foo'} onClickHandler={mockFn}/>);

    it('should be defined', () => {
        expect(select).toBeDefined();
    });
    it('should render correctly', () => {
        expect(select).toMatchSnapshot();
    });
});
