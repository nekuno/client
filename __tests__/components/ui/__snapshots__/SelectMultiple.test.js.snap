// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SelectMultiple component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectMultiple
    labels={
      Array [
        Object {
          "id": "lorem",
          "text": "Lorem",
        },
        Object {
          "id": "ipsum",
          "text": "Ipsum",
        },
        Object {
          "id": "lorem-ipsum",
          "text": "Lorem ipsum",
        },
        Object {
          "id": "sed-ut-perspiciatis",
          "text": "Sed ut perspiciatis",
        },
      ]
    }
    onClickHandler={[MockFunction]}
    title="Foo"
    values={
      Array [
        "lorem",
        "lorem-ipsum",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="chips"
      >
        <div
          className="title small"
        >
          Foo
        </div>
        <Chip
          onClickHandler={[Function]}
          selected={true}
          text="Lorem"
        />
        <Chip
          onClickHandler={[Function]}
          selected={false}
          text="Ipsum"
        />
        <Chip
          onClickHandler={[Function]}
          selected={true}
          text="Lorem ipsum"
        />
        <Chip
          onClickHandler={[Function]}
          selected={false}
          text="Sed ut perspiciatis"
        />
      </div>,
      "className": "selectMultiple",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title small"
          >
            Foo
          </div>,
          Array [
            <Chip
              onClickHandler={[Function]}
              selected={true}
              text="Lorem"
            />,
            <Chip
              onClickHandler={[Function]}
              selected={false}
              text="Ipsum"
            />,
            <Chip
              onClickHandler={[Function]}
              selected={true}
              text="Lorem ipsum"
            />,
            <Chip
              onClickHandler={[Function]}
              selected={false}
              text="Sed ut perspiciatis"
            />,
          ],
        ],
        "className": "chips",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Foo",
            "className": "title small",
          },
          "ref": null,
          "rendered": "Foo",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "lorem",
          "nodeType": "class",
          "props": Object {
            "color": undefined,
            "onClickHandler": [Function],
            "selected": true,
            "text": "Lorem",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ipsum",
          "nodeType": "class",
          "props": Object {
            "color": undefined,
            "onClickHandler": [Function],
            "selected": false,
            "text": "Ipsum",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "lorem-ipsum",
          "nodeType": "class",
          "props": Object {
            "color": undefined,
            "onClickHandler": [Function],
            "selected": true,
            "text": "Lorem ipsum",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "sed-ut-perspiciatis",
          "nodeType": "class",
          "props": Object {
            "color": undefined,
            "onClickHandler": [Function],
            "selected": false,
            "text": "Sed ut perspiciatis",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="chips"
        >
          <div
            className="title small"
          >
            Foo
          </div>
          <Chip
            onClickHandler={[Function]}
            selected={true}
            text="Lorem"
          />
          <Chip
            onClickHandler={[Function]}
            selected={false}
            text="Ipsum"
          />
          <Chip
            onClickHandler={[Function]}
            selected={true}
            text="Lorem ipsum"
          />
          <Chip
            onClickHandler={[Function]}
            selected={false}
            text="Sed ut perspiciatis"
          />
        </div>,
        "className": "selectMultiple",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title small"
            >
              Foo
            </div>,
            Array [
              <Chip
                onClickHandler={[Function]}
                selected={true}
                text="Lorem"
              />,
              <Chip
                onClickHandler={[Function]}
                selected={false}
                text="Ipsum"
              />,
              <Chip
                onClickHandler={[Function]}
                selected={true}
                text="Lorem ipsum"
              />,
              <Chip
                onClickHandler={[Function]}
                selected={false}
                text="Sed ut perspiciatis"
              />,
            ],
          ],
          "className": "chips",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Foo",
              "className": "title small",
            },
            "ref": null,
            "rendered": "Foo",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "lorem",
            "nodeType": "class",
            "props": Object {
              "color": undefined,
              "onClickHandler": [Function],
              "selected": true,
              "text": "Lorem",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "ipsum",
            "nodeType": "class",
            "props": Object {
              "color": undefined,
              "onClickHandler": [Function],
              "selected": false,
              "text": "Ipsum",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "lorem-ipsum",
            "nodeType": "class",
            "props": Object {
              "color": undefined,
              "onClickHandler": [Function],
              "selected": true,
              "text": "Lorem ipsum",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "sed-ut-perspiciatis",
            "nodeType": "class",
            "props": Object {
              "color": undefined,
              "onClickHandler": [Function],
              "selected": false,
              "text": "Sed ut perspiciatis",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
